### PR TITLE
Improve how expressions show up on the leaderboard

### PR DESCRIPTION
**Pull request by JosiasAurel**
Addresses issue https://github.com/hackclub/sinerider/issues/555

### CODE REVIEW

**Comment by grymmy** [comment]
lgtm - but could you do me a favor @JosiasAurel and give a brief one/two sentence ELI5 explanation of what the actual problem was and your solution to it?

@ivoinestrachan please review & merge this if you deem appropriate.

**Review by grymmy — APPROVED** [review]

**Comment by JosiasAurel** [comment]
@grymmy 
The problem is that the MathJax component requires that expressions start with \\\\( and end with \\\\) but that isn't in the expression we get back from airtable so it would fail to parse the math. There's no need for `<MathJax.Node />` in a Provider since a MathJax context is already included in the Next app.

There's also the problem with escape sequences in the expression like `\t` which was a problem so I made a raw string so that they don't get escaped.

The rest is just making sure the score expressions get converted to a mathjax compatible one before they are rendered

**Comment by grymmy** [comment]
> @grymmy 
> The problem is that the MathJax component requires that expressions start with \\\\( and end with \\\\) but that isn't in the expression we get back from airtable so it would fail to parse the math. There's no need for `<MathJax.Node />` in a Provider since a MathJax context is already included in the Next app.
> 
> There's also the problem with escape sequences in the expression like `\t` which was a problem so I made a raw string so that they don't get escaped.
> 
> The rest is just making sure the score expressions get converted to a mathjax compatible one before they are rendered

Thanks I feel we may have this same issue elsewhere, helps to have this context
